src/groovy/org/transmart/FederatedUserDetailsService.groovy

```diff
@@ -111,6 +112,13 @@ public class FederatedUserDetailsService implements SAMLUserDetailsService {
 
             AuthUser newUser = AuthUser.createFederatedUser(federatedId,
                     username, realName, email, sessionFactory.currentSession);
+
+            if (attributeConfig.newUserAuthorities) {
+                // if new user authorities specified then replace default authorities
+                newUser.authorities.collect{}.each { newUser.removeFromAuthorities(it) }
```

Comment:
Unsuccessfully it will not work, because in this case objects will not be deleted from database. You can find more details [here] (http://spring.io/blog/2010/07/02/gorm-gotchas-part-2/) in "Deleting children" section. Better solution in this case is adding:
static mapping = {
    authorities cascade: "all-delete-orphan"
}
but AuthUser definition located in search-domain plug-in, so it is hard to submit all these changes in one pull request.